### DISCUSSION TITLE
Ability to get existing instead of creating new one

### DISCUSSION THREAD

**Discussion by bojmaliev** [ash-jc-allen/short-url#145] [Q&A]
Hello,

I come to situation where i don't want to create and duplicate my urls in my database, but i want to create or get the existing short link for certain destination url. 

Thank you in advance

**Comment by ash-jc-allen** [comment]
Hey!

The `ShortURL` class is just a Laravel `Model` class, which means that you can have access to the usual Model and Eloquent methods.

There are also some helper methods on the `ShortURL` model that you can use to check if a short URL already exists, such as:

```php
$shortURL = \AshAllenDesign\ShortURL\Models\ShortURL::findByKey('abc123');
```

and

```php
$shortURLs = \AshAllenDesign\ShortURL\Models\ShortURL::findByDestinationURL('https://destination.com');
```

So, you might want to check whether a short URL exists. If it does, you can manually update that short URL. If it doesn't, you can create the new short URL.

Hopefully this helps a bit? 😄
